Source/WebKit/NetworkProcess/cocoa/NetworkTaskCocoa.mm

```diff
@@ -142,18 +142,22 @@ static bool shouldCapCookieExpiryForThirdPartyIPAddress(const WebCore::IPAddress
 }
 
 #if HAVE(ALLOW_ONLY_PARTITIONED_COOKIES)
-void NetworkTaskCocoa::setCookieTransformForFirstPartyRequest(const WebCore::ResourceRequest& request)
+void NetworkTaskCocoa::setCookieTransformForThirdPartyRequest(const WebCore::ResourceRequest& request, IsRedirect isRedirect)
 {
     if (!request.isThirdParty())
         return;
 
-    ASSERT(!task()._cookieTransformCallback);
-    if (!!task()._cookieTransformCallback)
-        return;
-
     if (!Quirks::needsPartitionedCookies(request))
         return;
 
+#ifndef NDEBUG
+    ASSERT(!task()._cookieTransformCallback || isRedirect == IsRedirect::Yes);
```

Comment:
I think this can just use `ASSERT_UNUSED` instead of having to use ifdefs right

Reply:
I probably knew this at some point :)